prow/cmd/entrypoint/main.go

```diff
@@ -25,9 +28,61 @@ import (
 	"k8s.io/test-infra/prow/pod-utils/options"
 )
 
+const (
+	copyModeOnlyEnvVar = "COPY_BINARY_ONLY"
+
+	defaultCopyDst = "/tools/entrypoint"
+)
+
+var defaultCopySrc = []string{"/ko-app/entrypoint", "/entrypoint"}
+
+// copy copies entrypoint binary from source to destination. This is because
+// entrypoint image operates in two different modes:
+// 1) entrypoint container: copy the binary to shared mount drive `/tools`
+// 2) test container(s): use `/tools/entrypoint` as entrypoint, for collecting
+//    logs and artifacts.
+func copy(srcs []string, dst string) error {
+	var info fs.FileInfo
+	var err error
+	var src string
+	for _, src = range srcs {
+		info, err = os.Stat(src)
+		if err != nil {
+			if !os.IsNotExist(err) {
+				return fmt.Errorf("stats '%s': %w", src, err)
+			}
+			// if /ko-app/entrypoint doesn't exist, fall back to /entrypoint
+			// TODO(chaodaiG): disable this feature as soon as prow images are
+			// not published by bazel
+			continue
+		}
+		// if the file exist then use it
+		break
+	}
+	// None of the files exist
+	if err != nil {
+		return fmt.Errorf("neither files in %v exist", srcs)
+	}
+	body, err := ioutil.ReadFile(src)
+	if err != nil {
+		return fmt.Errorf("read '%s': %w", src, err)
+	}
+	if err := ioutil.WriteFile(dst, body, info.Mode()); err != nil {
+		return fmt.Errorf("write '%s': %w", dst, err)
+	}
+	return nil
+}
+
 func main() {
 	logrusutil.ComponentInit()
 
+	if os.Getenv(copyModeOnlyEnvVar) == "true" {
```

Comment:
Is there a reason to not use a flag? The advantage of using a flag is that it is discoverable by calling `--help` which isn't the case for an env var

Reply:
good point, updated